app/views/unattended/provisioning_templates/snippet/saltstack_minion.erb

```diff
@@ -20,4 +23,4 @@ log_level: warning
 # * {'roles': ['webserver', 'frontend']}
 #
 %>
-grains: <%= host_param('salt_grains') ? host_param('salt_grains') : '{}' %>
+grains: <%= to_json(@host.respond_to?(:derive_salt_grains) ? @host.derive_salt_grains(use_autosign: true) : {}) %>
```

Comment:
Another solution to avoid the `derive_salt_grains` is undefined would be:
```suggestion
<% if plugin_present?('foreman_salt') -%>
grains: <%= to_json(@host.respond_to?(:derive_salt_grains) ? @host.derive_salt_grains(use_autosign: true) : {}) %>
<% else -%>
grains: <%= host_param('salt_grains') ? host_param('salt_grains') : '{}' %>
 <% end -%>
 
 Whether this is something you want, is a different question ;)
```

Reply:
Yeah we need to do it this way, we do the same for Katello.